vkcs/kubernetes/resource_cluster_test.go

```diff
@@ -261,3 +396,81 @@ resource "vkcs_kubernetes_cluster" "update" {
   ]
 }
 `
+
+const testAccKubernetesClusterRegionalBasic = `
+{{ .TestAccKubernetesNetworkingBase }}
+{{ .TestAccKubernetesClusterBase }}
+
+data "vkcs_compute_flavor" "base" {
+  name = "Standard-2-8-50"
+}
+
+resource "vkcs_kubernetes_cluster" "basic" {
+  name                = "{{ .ClusterName }}"
+  cluster_template_id = data.vkcs_kubernetes_clustertemplate.base.id
+  master_flavor       = data.vkcs_compute_flavor.base.id
+  master_count        = 3
+  network_id          = vkcs_networking_network.base.id
+  subnet_id           = vkcs_networking_subnet.base.id
+  floating_ip_enabled = false
+  cluster_type 		  = "regional" 
+
+  depends_on = [
+    vkcs_networking_router_interface.base,
+  ]
+}
+`
+
+const testAccKubernetesClusterRegionalUpdate = `
+{{ .TestAccKubernetesNetworkingBase }}
+{{ .TestAccKubernetesClusterBase }}
+
+data "vkcs_compute_flavor" "base" {
+  name = "Standard-4-12"
+}
+
+resource "vkcs_kubernetes_cluster" "update" {
+  name                = "{{ .ClusterName }}"
+  cluster_template_id = data.vkcs_kubernetes_clustertemplate.base.id
+  master_flavor       = data.vkcs_compute_flavor.base.id
+  master_count        = 3
+  network_id          = vkcs_networking_network.base.id
+  subnet_id           = vkcs_networking_subnet.base.id
+  floating_ip_enabled = false
+  cluster_type 		  = "regional"
+
+  labels = {
+	etcd_volume_size = "{{ .EtcdVolumeSize }}"
+	clean_volumes    = "{{ .CleanVolumes }}"
+	kube_log_level   = "{{ .KubeLogLevel }}"
+  }
+
+  depends_on = [
+    vkcs_networking_router_interface.base,
+  ]
+}
+`
+
+const testAccKubernetesClusterRegionalResize = `
+{{ .TestAccKubernetesNetworkingBase }}
+{{ .TestAccKubernetesClusterBase }}
+
+data "vkcs_compute_flavor" "base" {
+  name = "{{ .FlavorName }}"
+}
+
+resource "vkcs_kubernetes_cluster" "basic" {
+  name                = "{{ .ClusterName }}"
+  cluster_template_id = data.vkcs_kubernetes_clustertemplate.base.id
+  master_flavor       = data.vkcs_compute_flavor.base.id
+  master_count        = 3
+  network_id          = vkcs_networking_network.base.id
+  subnet_id           = vkcs_networking_subnet.base.id
+  floating_ip_enabled = false
+  cluster_type 		  = "regional" 
+
+  depends_on = [
+    vkcs_networking_router_interface.base,
+  ]
+}
+`
```

Comment:
I will leave comment here but it relates to all tests changes. I don't see a reason to duplicate all tests for regional clusters, it's too much, these tests are heavy-weighted. We have test for creation in examples, and I don't see any special logic to duplicate even it for now. Also, there is no reason to have separate names, and etc., despite it's named "cluster_type", it's the same cluster with new piece of functionality.

Reply:
Do I need to delete all tests with regional clusters or can I keep the basic one?

Reply:
You can just set it type regional in test for cluster update, for example.